spotify_tensorflow/dictionary_endpoint.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import logging
+import sys
+import timeit
+from collections import OrderedDict
+
+import six
+import tensorflow as tf
+import numpy as np
+
+from .dataset import Datasets
+
+
+FLAGS = tf.flags.FLAGS
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DictonaryEndpoint(object):
```

Comment:
What would you think of instead of exposing these:
 * `DictionaryEndpoint`
 * `DataFrameEndpoint`

as separate classes, instead expose it via `Datasets` API, so it would be something like:

```
Datasets.dict.read_dataset(...)
Datasets.dict.batch_iterator(...)
```

and for dataframe:

```
Datasets.dataframe.read_dataset(...)
Datasets.dataframe.batch_iterator(...)
```

and we would mark both `DictionaryEndpoint` and `DataFrameEndpoint` as private. This should simplify the API i believe.

What do you think? also pinging @yonromai

Reply:
Refactored in this way.  Let me know what you think.